devito/operator.py

```diff
@@ -165,17 +145,17 @@ def arguments(self, **kwargs):
                     new_params[orig_child.name] = new_child
         kwargs.update(new_params)
 
-        for ta in self.tensor_args:
-            assert(ta.verify(kwargs.pop(ta.name, None)))
-
-        for d in self.dims:
+        # Derivation
+        for i in self.parameters:
+            if i.is_TensorArgument:
+                assert(i.verify(kwargs.pop(i.name, None)))
+            else:
+                i.verify(kwargs.pop(i.name, None))
```

Comment:
This is changing the tree structure of the arguments derivation. Why is this required? Especially in this PR?

Reply:
Can you provide an example of what gets broken here? As all tests are passing, hence probably there's a flaw in our testing infrastructure.

Reply:
The arguments derivation is a WIP, including tests. 

- You removed the tree structure by putting it all in one loop
- Scalar Arguments aren't being called any more
- Dimensions returned to the assumption that there is only one argument per dimension. Undoing my previous cleanup regarding this

Reply:
Again, I need an example, or **better**, a test to catch what's breaking here, otherwise the only option is to go ahead and "fix" this in a second moment, when the test will be available

My concerns follow:
* ``self.parameters`` includes both tensor and scalar aguments. So ``ScalarArgument``s are actually being called (``else`` branch).
* I don't see how I could have altered the tree structure, since you also had several loops
* I don't understand the point about dimensions, as it seems the same loop to me ?

I totally understand the arguments stuff was WIP; but still, this can't be my fault if no tests caught this "issue" (?)